platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TimeLimitedMatcher.java

```diff
@@ -0,0 +1,171 @@
+ * case. It is incumbent on the caller to handle the runtime exception.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *   try {
+ *     Pattern pattern = Pattern.compile("bc");
+ *     TimeLimitedMatcher matcher = TimeLimitedMatcher.create(pattern, "abc123");
+ *     if (matcher.matches()) {
+ *       // do something with the match
+ *     }
+ *   } catch (TimeLimitedMatcher.TimeoutException e) {
+ *     // handle exception
+ *   }
+ * </pre>
+ */
+public class TimeLimitedMatcher {
+  // Limit the lifetime of this sequence to five seconds by default.
+  private static final Duration DEFAULT_DURATION = Duration.ofSeconds(5L);
+
+  private final Matcher matcher;
+
+  private boolean matched;
+
+  /**
+   * Creates a new Time-limited matcher instance to search for the regular expression {@code
+   * pattern} against the {@code inputString}.
+   *
+   * <p>This method executes the search before returning, throwing a {@link TimeoutException} if the
+   * regex search runs longer than {@link #DEFAULT_DURATION}.
+   *
+   * @param pattern RegEx search pattern
+   * @param inputString String to match against.
+   * @return TimeLimitedMatcher instance to use for reporting out match details
+   */
+  public static TimeLimitedMatcher create(Pattern pattern, CharSequence inputString) {
+    return create(pattern, inputString, DEFAULT_DURATION);
+  }
+
+  /**
+   * Creates a new Time-limited matcher instance to search for the regular expression {@code
+   * pattern} against the {@code inputString}.
+   *
+   * <p>This method executes the search before returning, throwing a {@link TimeoutException} if the
+   * regex search runs longer than {@code duration}.
+   *
+   * @param pattern RegEx search pattern
+   * @param inputString String to match against.
+   * @param duration the length of time to continue processing the regex before failing
+   * @return TimeLimitedMatcher instance to use for reporting out match details
+   */
+  public static TimeLimitedMatcher create(
+      Pattern pattern, CharSequence inputString, Duration duration) {
+    Matcher matcher = pattern.matcher(new TLCharSequence(inputString, calculateEndTime(duration)));
+    return new TimeLimitedMatcher(matcher);
+  }
+
+  /**
+   * Returns the input subsequence matched by the previous match.
+   *
+   * @return The (possibly empty) subsequence matched by the previous match, in string form
+   * @see Matcher#group()
+   */
+  public String group() {
+    return matcher.group();
+  }
+
+  /**
+   * Returns the input subsequence captured by the given group during the previous match operation.
+   *
+   * @param group The index of a capturing group in this matcher's pattern
+   * @return The (possibly empty) subsequence captured by the group during the previous match, or
+   *     null if the group failed to match part of the input
+   * @see Matcher#group(int)
+   */
+  public String group(int group) {
```

Comment:
✏️ Should include other methods such as `groupCount()`,  `group(String name)`,  `toString()` (for debugging).